kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -99,75 +98,82 @@ grCollectionsPanel.controller('GrNodeCtrl',
-            remove$.withLatestFrom(ctrl.selectedImages$, (path, images) => ({path, images}));
+        ctrl.addImagesToCollection = () => {
+            ctrl.saving = true;
+            add$.onNext(ctrl.node.data.fullPath);
+        };
 
-    ctrl.removeImagesFromCollection = () => {
-        ctrl.removing = true;
-        remove$.onNext(ctrl.node.data.data.pathId);
-    };
+        subscribe$($scope, pathWithImages$, ({path, images}) => {
+           collections.addToCollectionUsingImageResources(images, path)
+           .then(() => ctrl.saving = false);
+        });
 
-    subscribe$($scope, pathToRemoveWithImages$, ({path, images}) => {
-        collections.batchRemove(images, path).then(() => ctrl.removing = false);
-    });
+        const remove$ = new Rx.Subject();
+        const pathToRemoveWithImages$ =
+                remove$.withLatestFrom(selectedImages$, (path, images) => ({path, images}));
 
-    inject$($scope, hasImagesSelected$, ctrl, 'hasImagesSelected');
+        ctrl.removeImagesFromCollection = () => {
+            ctrl.removing = true;
+            remove$.onNext(pathId);
+        };
 
-    $scope.$watch('ctrl.showChildren', onValChange(show => {
-        collectionsTreeState.setState(pathId, show);
-    }));
+        subscribe$($scope, pathToRemoveWithImages$, ({path, images}) => {
+            collections.batchRemove(images, path).then(() => ctrl.removing = false);
+        });
+
+        inject$($scope, hasImagesSelected$, ctrl, 'hasImagesSelected');
+
+        ctrl.isSelected = selectedCollections.some(col => {
+            return angular.equals(col, pathId);
+        });
+
+        ctrl.hasCustomSelect = !! grCollectionTreeCtrl.onSelect;
 
-    ctrl.isSelected = ctrl.selectedCollections.some(col => {
-        return angular.equals(col, ctrl.node.data.data.pathId);
-    });
+        ctrl.select = () => {
+            grCollectionTreeCtrl.onSelect({$collection: ctrl.node.data.data.path});
+        };
 
+    };
 }]);
 
 grCollectionsPanel.directive('grNode', ['$parse', '$compile', function($parse, $compile) {
     const templateString = `<gr-nodes
                                 ng:if="ctrl.showChildren && ctrl.node.data.children.length > 0"
-                                gr:selected-images="ctrl.selectedImages$"
-                                gr:selected-collections="ctrl.selectedCollections"
-                                gr:editing="ctrl.editing"
-                                gr:nodes="ctrl.node.data.children"></gr-nodes>`;
+                                gr:nodes="ctrl.node.data.children"
+                                ></gr-nodes>`;
     return {
         restrict: 'E',
+        require: ['grNode', '^^grCollectionTree'],
         scope: {
             node: '=grNode',
-            nodeList: '=grNodeList',
-            editing: '=grEditing',
-            selectedImages$: '=grSelectedImages',
-            selectedCollections: '=grSelectedCollections'
+            nodeList: '=grNodeList'
```

Comment:
`nodeList` is a list of 'sibling' nodes, which is used when a collection is deleted. So, think it does need to be passed down recursively.